### ISSUE TITLE
The isatty check is always done on stdout, even if the output file is different

### ISSUE THREAD

**Issue by avdstaaij** [termcolor/termcolor#41]
It seems that termcolor always checks `sys.stdout` for being a TTY, instead of the file you are actually outputting to.
A consequence of this is that termcolor will not color output to `sys.stderr` when a program is used in a pipe. This is an unfortunate limitation.

It makes sense that `colored` always checks `sys.stdout`, since it has no way of knowing what the output file will be. However, I would like to request there to be some way of explicitly specifying the intended output file, perhaps via an optional `output_file` argument.

For `cprint`, this is arguably a bug. I believe it could be fixed by looking for the `file` argument in `kwargs` and then checking that instead of `sys.stdout`.

**Comment by hugovk** [comment]
How does PR #56 look?

**Comment by hugovk** [comment]
Fixed by https://github.com/termcolor/termcolor/pull/56.

**Comment by avdstaaij** [comment]
Apologies, I completely missed this notification.

Unfortunately, this does not seem to be fixed as of commit 825ad6508bbca86a945e841d8c738ad2c5276650.

My test scenario:

I installed `termcolor` in a virtual environment using `pip install "termcolor @ git+https://github.com/termcolor/termcolor@825ad6508bbca86a945e841d8c738ad2c5276650"`.
I created the following executable file `test.py`:
```python
#!/usr/bin/env python3

import sys
import termcolor

termcolor.cprint("cprint", "red", file=sys.stderr)
print(termcolor.colored("colored", "red"), file=sys.stderr)
```
I ran:
```
./test.py
./test.py > /dev/null
```

Expected/desired result:
The first three output lines are red, and the fourth preferably as well.

Observed result:
The first two output lines are red, the second two are white.
